src/libraries/System.Net.WebProxy/src/System/Net/WebProxy.cs

```diff
@@ -219,7 +224,20 @@ private sealed class ChangeTrackingArrayList : ArrayList
 
             public ChangeTrackingArrayList(ICollection c) : base(c) { }
 
-            public bool IsChanged { get; set; }
+            private volatile bool _isChanged;
+
+            public bool IsChanged
+            {
+                get
+                {
+                    return _isChanged;
+                }
+
+                set
+                {
+                    _isChanged = value;
+                }
+            }
```

Comment:
```suggestion
            public volatile bool IsChanged;
```

Since these are already internal implementation details, we can just make the field public.